src/main/java/fi/aalto/cs/intellij/activities/RequiredPluginsCheckerActivity.java

```diff
@@ -0,0 +1,213 @@
+        disabledPluginDescriptors.add(descriptor);
+      }
+    }
+
+    if (missingPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsInstallation(missingPluginDescriptors);
+    } else if (disabledPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsEnabling(disabledPluginDescriptors);
+    }
+  }
+
+  /**
+   * Notify with an option to enable all the required plugins.
+   */
+  private void notifyAndSuggestPluginsEnabling(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
+        "A+",
+        "A+",
+        "Some plugins must be and enabled for the A+ plugin to work properly "
+            + getPluginsNamesString(descriptors) + ".",
+        NotificationType.WARNING);
+
+    notification.addAction(new NotificationAction(
+        "Enable the required plugin(s) (" + getPluginsNamesString(descriptors) + ").") {
+
+      /**
+       * Activate all the required plugins.
+       */
+      @Override
+      public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+        descriptors.forEach(descriptor -> Objects
+            .requireNonNull(PluginManager.getPlugin(descriptor.getPluginId())).setEnabled(true));
+        notification.expire();
+      }
+    });
+
+    Notifications.Bus.notify(notification);
+  }
+
+  /**
+   * Notify with an option to install all the required plugins and suggest restart.
+   */
+  private void notifyAndSuggestPluginsInstallation(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
+        "A+",
+        "A+",
+        "Additional plugin(s) must be installed and enabled for the A+ plugin to work "
+            + "properly (" + getPluginsNamesString(descriptors) + ").",
+        NotificationType.WARNING);
+
+    notification.addAction(new NotificationAction(
+        "Install missing (" + getPluginsNamesString(descriptors) + ") plugin(s).") {
+
+      /**
+       * Install the missing plugins and propose a restart.
+       */
+      @Override
+      public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+        descriptors.forEach(descriptor -> {
+          try {
+            PluginDownloader pluginDownloader = PluginDownloader.createDownloader(descriptor);
+            pluginDownloader.prepareToInstall(new BgProgressIndicator());
+            pluginDownloader.install();
+          } catch (IOException ex) {
+            logger.error("Could not install plugin" + descriptor.getName() + ".", ex);
+          }
+        });
+        notification.expire();
+        PluginManagerConfigurable
+            .shutdownOrRestartApp("Plugins required for A+ course are now installed");
+      }
+    });
+
+    Notifications.Bus.notify(notification);
+  }
+
+  /**
+   * Join plugin descriptor names with a comma.
+   */
+  @NotNull
+  private static StringJoiner getPluginsNamesString(List<IdeaPluginDescriptor> descriptors) {
```

Comment:
Check Collectors.joining(","), e.g. something along the lines of:
descriptors.stream().map(IdeaPluginDescriptor::getName).collect(Collectors.joining(", "));